clients/src/test/java/org/apache/kafka/common/utils/Serializer.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license agreements. See the NOTICE
+ * file distributed with this work for additional information regarding copyright ownership. The ASF licenses this file
+ * to You under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the
+ * License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package org.apache.kafka.common.utils;
+
+import java.io.IOException;
+import java.io.ByteArrayOutputStream;
+import java.io.ByteArrayInputStream;
+import java.io.ObjectOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.ObjectInputStream;
+
+public class Serializer {
+
+    public byte[] serialize(Object toSerialize) throws IOException {
+        byte[] byteArray = null;
+        ByteArrayOutputStream arrayOutputStream = new ByteArrayOutputStream();
+        ObjectOutputStream ooStream = new ObjectOutputStream(arrayOutputStream);
+
+        ooStream.writeObject(toSerialize);
+        byteArray = arrayOutputStream.toByteArray();
+
+        ooStream.close();
+        arrayOutputStream.close();
+        return byteArray;
+    }
+
+    public Object deserialize(byte[] byteArray) throws IOException, ClassNotFoundException {
+        ByteArrayInputStream arrayInputStream = new ByteArrayInputStream(byteArray);
+        ObjectInputStream inputStream = new ObjectInputStream(arrayInputStream);
+
+        Object deserializedObject = inputStream.readObject();
+
+        inputStream.close();
+        arrayInputStream.close();
+
+        return deserializedObject;
+    }
+
+    public Object deserialize(String fileName) throws IOException, ClassNotFoundException {
```

Comment:
This and the other deserialize method have a lot of duplicated code. We could have a method that takes an InputStream with the shared logic.